### PR TITLE
PP-8026 - Refactor performance report with transaction summary table

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- Refactors /performance-report to use transaction_summary table
- Since there are no used methods in PerformanceReportDao this can be safely deleted as well as removing the dependency injection provided in LedgerModule
- The corresponding toolbox PR will have to be merged to use LocalDate rather than ZonedDate

### CODE REVIEW

File: src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java [hunk 1]
```diff
@@ -59,6 +63,17 @@
...
             "ORDER BY t.gateway_account_id, year, month";
 
+    private static final String PERFORMANCE_REPORT = "SELECT COALESCE(SUM(t.no_of_transactions),0) AS volume, " +
+            "COALESCE(SUM(t.total_amount_in_pence), 0) AS total_amount, " +
+            "COALESCE(SUM(t.total_amount_in_pence)/SUM(t.no_of_transactions), 0) AS avg_amount " +
+            "FROM transaction_summary t " +
+            "WHERE t.type= 'PAYMENT' " +
+            "AND t.live= TRUE";
+
+    private static final String WITH_STATE = " and t.state=:state";
```
Comment (SandorArpa) [inline src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java:73]: Shouldn't it be all uppercase AND to be consistent with the SQL?

File: src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java [hunk 2]
```diff
@@ -59,6 +63,17 @@
...
+
+    private static final String WITH_DATE_RANGE = " and t.transaction_date between :startDate and :toDate";
```
Comment (SandorArpa) [inline src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java:75]: As above, all uppercase AND?

File: src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java [hunk 3]
```diff
@@ -59,6 +63,17 @@
             "GROUP BY t.gateway_account_id, year, month " +
             "ORDER BY t.gateway_account_id, year, month";
 
+    private static final String PERFORMANCE_REPORT = "SELECT COALESCE(SUM(t.no_of_transactions),0) AS volume, " +
```
Comment (SandorArpa) [inline src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java:66]: Perhaps a bit more meaningful name, like `PERFORMANCE_REPORT_QUERY`?

**Review by SandorArpa — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java (hunk 1 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java:73]: Done.

File: src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java (hunk 3 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java]: Done.

**Review by SandorArpa — APPROVED** [review]
LOKTM

**Review by SandorArpa — APPROVED** [review]
LOKTM

**Pull request merged**